src/parser/Parser/Parser.hs

```diff
@@ -531,7 +539,7 @@ optionalTypeParameters = option [] (brackets $ commaSep1 modedTypeVar)
 
 typedef :: EncParser Typedef
 typedef = do
-  typedefmeta <- meta <$> getPosition
+  typedefmeta <- meta . newPos <$> getPosition
```

Comment:
It's a question of where to put the complexity. In the parser, we get a `SourcePos` to create a `Meta` from. In the rest of the compiler, we create new `Meta`s by getting the `Position`s from other `Meta`s. We went with this solution since it kept the complexity local to a single module. An alternative would be to expose two constructors for `Meta`s, `meta` and `metaRaw` or something, but we would still need to use different functions in different places. Is this better in your opinion?